### ISSUE TITLE
ACF PRO test version 5.7.11

### ISSUE THREAD

**Issue by elliotcondon** [AdvancedCustomFields/acf#125]
Hi all 👋,

The next version of ACF PRO is nearly complete and contains a lot of refactored code that needs to be tested. This version starts a long process of tidying up the entire ACF codebase which I will continue throughout the year.

Specifically, this version contains refactored code for the low level functions used throughout the plugin:
- Field functions
- Field group functions
- Meta functions
- Value functions

In the process of refactoring, I have addressed many of the cache issues that have plagued ACF. As a result, ACF now correctly caches results to minimize database calls. I suggest using the [Query Monitor plugin](https://en-au.wordpress.org/plugins/query-monitor/) to view database calls in order to test this.

I will be very grateful if you can spend some time testing out the "5.7.11-beta1" version available from your [account online](https://www.advancedcustomfields.com/my-account
). Just click the "See all versions" link alongside your license for all the available version downloads.

Areas to focus are:
- Editing a field group
- Editing a post, user, comment, widget, term, options page
- Using the "Local JSON" feature
- Syncing, exporting and importing field groups
- Testing Memcached and reviewing SQL queries

Thanks in advance for any and all help you are able to provide. I'm extremely proud of the work that has gone into this version and I can't wait to release this update.

Please post any issues found here on GitHub 👍.

🎉🎉🎉

Thanks
Elliot

**Comment by johanhermansson** [comment]
Looking forward to testing this 👍 I hope it will fix the field sorting problems I've had with fields within Flexible content when synchronising them from the local JSON files.

**Comment by matthiaswagnerwy** [comment]
hy,
i just installed 5.7.11 and found out that slashes are added to single and double quotes in my textarea-field in an options page. i tried to figure out if other plugins could be the reason, but it looks like this comes from acf.
any hints on this?

matt

**Comment by jossoway** [comment]
@matthiaswagnerwy @elliotcondon I am encountering the same issue since upgrading a site staging environment to 5.7.11. I rolled back to 5.7.10 and the issue disappeared.

**Comment by sethwhitaker** [comment]
Did this get pushed to Wordpress plugin database early? This update broke all my flexible content tab areas. We have a pretty complex page builder with ACF and flex content fields which has gotten all sorts of messed up with `5.7.11`

**Comment by elliotcondon** [comment]
@matthiaswagnerwy Thanks. I've just replicated and fixed this bug.

**Comment by elliotcondon** [comment]
@sethwhitaker Yes. ACF PRO 5.7.11 has been released. If you are experiencing an issue with the flexible content field, please help me replicate the issue and I'll be sure to get it fixed ASAP.

**Comment by kinjaldalwadi** [comment]
Hello,
I have used ACF pro , it was working in old version but not working in latest wordpress version. Repeater and flexible content field are not working as require. Can you please help?

**Comment by sethwhitaker** [comment]
Going to be hard to get you to replicate the issue because we have a whole system in place for dynamically generating flex content layout fields in a page builder from php arrays. I would need to give you access to our system to help you track it down.

The interesting thing though, is that I downloaded the 5.8 beta 3 and it was back to normal. So there is definitely something weird going on in 5.7.11 with tabs. Because its the part of our class that deals with dynamically setting up some tab sections by injecting some fields into the default fields array which is acting wonky. The First tab that im injecting is not getting added, but like i said, its totally fine in 5.7.10 and 5.8b3. 

Just looking at a changelog from 5.7.10 to 5.7.11 it just looked like you made a lot of changes for such a minor version bump.

**Comment by elliotcondon** [comment]
Hi @sethwhitaker 

We refactored a lot of code in 5.7.11 but did not intend to make any breaking changes. I'm motivated to find and fix these issues, so if you are able to, please create a simple environment to replicate the issue. Perhaps there is a bug with one of the filters you are using? OR perhaps a bug in the "Local Fields" logic.

**Comment by sethwhitaker** [comment]
I know its not helpful for you, but im not going to be able to set up a test environment because its company code. Doing more digging, I do think the issue lies somewhere in the add_local_field_group logic. We have a custom wrapper class around that function where we inject our custom tab fields and thats the location ive pinpointed where its breaking in 5.7.11. 

Have the changes you made in `5.7.11` also made it into `5.8b`? could the reason that 5.8 is working fine because its based off of `5.7.10`?

**Comment by elliotcondon** [comment]
@sethwhitaker ACF PRO 5.8.0-beta3 does not contain any of the updates from 5.7.11.
I intend to release a 5.8.0-beta4 shortly to include all the new 5.7.x changes from the past few months.

**Comment by wcjonathan** [comment]
I have an issue when updating to version 5.7.11+ where ACF Gallery fields stop working and outputting any previously added images (not all images though strangely enough). Reverting back to version 5.7.10 brings back the images. I have tried 5.7.12 this morning but it's still broken.

**Comment by sethwhitaker** [comment]
The changelog on `5.7.12` looked promising for the fixes you made, but it still it broken for us. Ill do my best this weekend to try to set up a simplified test environment for you to look at.

**Comment by elliotcondon** [comment]
@wcjonathan Thanks for the feedback. Can you please open a support ticket with a detailed description to replicate the issue? https://www.advancedcustomfields.com/contact/

**Comment by chuckreynolds** [comment]
upgraded from `5.7.10` -> `5.7.12` and so far it looks like that fixed the issues I was having with json config not being saved and certain parts on the front-end not being rendered using object cache. _(using [simple-cache](https://wordpress.org/plugins/simple-cache/) plugin)_
...so far. lookin good.

**Comment by alicam** [comment]
@elliotcondon Did this all make its way into PRO v5.8 ?? I can't immediately tell if it did...

A

**Comment by elliotcondon** [comment]
Hi @alicam 

Yes. These improvements were included in the 5.7.11 update :)
